crates/db_views/src/post_view.rs

```diff
@@ -320,6 +321,16 @@ fn queries<'a>() -> Queries<
       query = query.filter(post_actions::hidden.is_null());
     }
 
+    if o.hide_media.unwrap_or(o.local_user.hide_media()) {
+      query = query.filter(
+        post::url_content_type.is_null().or(
+          post::url_content_type
+            .not_like("image/%")
+            .and(post::url_content_type.not_like("video/%")),
+        ),
+      );
+    }
```

Comment:
This is just personal preference, but could you change it to this instead (makes more logical sense to me since its doing !hide, rather than show):

```rust
query = query.filter(not(
  post::url_content_type.is_not_null().and(
    post::url_content_type
      .like("image/%")
      .or(post::url_content_type.like("video/%")),
  ),
))
```